app/src/main/java/org/oppia/app/player/state/ImageRegionSelectionInteractionView.kt

```diff
@@ -0,0 +1,91 @@
+package org.oppia.app.player.state
+
+import android.content.Context
+import android.os.Handler
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.databinding.BindingAdapter
+import org.oppia.app.model.ImageWithRegions
+import org.oppia.app.utility.ClickableAreasImage
+import org.oppia.app.utility.OnClickableAreaClickedListener
+import org.oppia.util.accessibility.CustomAccessibilityManager
+import org.oppia.util.gcsresource.DefaultResourceBucketName
+import org.oppia.util.parser.ExplorationHtmlParserEntityType
+import javax.inject.Inject
+
+class ImageRegionSelectionInteractionView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0
+) : AppCompatImageView(context, attrs, defStyleAttr) {
+
+  private var isAccessibilityEnabled: Boolean = false
+  private var clickableAreas: List<ImageWithRegions.LabeledRegion> = emptyList()
+
+  @Inject
+  lateinit var accessibilityManager: CustomAccessibilityManager
+
+  @Inject
+  @field:ExplorationHtmlParserEntityType
+  lateinit var entityType: String
+
+  @Inject
+  @field:DefaultResourceBucketName
+  lateinit var resourceBucketName: String
+
+  private lateinit var entityId: String
+
+  fun setEntityId(entityId: String) {
+    this.entityId = entityId
+  }
+
+  fun setClickableAreas(clickableAreas: List<ImageWithRegions.LabeledRegion>) {
+    this.clickableAreas = clickableAreas
+  }
+
+  fun getClickableAreas(): List<ImageWithRegions.LabeledRegion> {
+    return clickableAreas
+  }
+
+  fun isAccessibilityEnabled(): Boolean{
+    return isAccessibilityEnabled
+  }
+//
+//  override fun onAttachedToWindow() {
+//    super.onAttachedToWindow()
+//    FragmentManager.findFragment<InjectableFragment>(this).createViewComponent(this).inject(this)
+//    isAccessibilityEnabled = accessibilityManager.isScreenReaderEnabled()
+//
+}
+
+/** Bind ItemTouchHelperSimpleCallback with RecyclerView for a [DragDropSortInteractionView] via data-binding. */
+@BindingAdapter(value = ["onRegionClicked", "overlayView"], requireAll = false)
+fun setRegionClickToImageView(
+  imageRegionSelectionInteractionView: ImageRegionSelectionInteractionView,
+  onClickableAreaClickedListener: OnClickableAreaClickedListener,
+  parentView: FrameLayout
+) {
+  val area = ClickableAreasImage(
+    imageRegionSelectionInteractionView,
+    parentView,
+    onClickableAreaClickedListener
+  )
+  Handler().postDelayed({
+    area.addViews(imageRegionSelectionInteractionView.isAccessibilityEnabled())
+  }, 100)
```

Comment:
Why have a delay? I'm concerned this will add flakiness & churn.

Reply:
if we do this immediately the image rect is not properly calculated and it returns the absolute width and height instead of the relative width height of image